go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -2173,3 +2186,86 @@ func generateIPFragmentReassemblyFlow(ofPortPhys string) []string {
 
 	return flows
 }
+
+// deleteStaleMasqueradeResources() function removes stale Linux resources when
+// config.Gateway.V4MasqueradeSubnet or config.Gateway.V4MasqueradeSubnet gets
+// changed at day 2.
+func deleteStaleMasqueradeResources(bridgeName string) error {
```

Comment:
@qinqon I was trying hard to not hardcode anything which we are going to obsolete. Instead I preferred cleaning resources after deriving IPs based on the masquerade address attached to br-ex link.
I would request you to check my approach once. If my approach is too dirty then I will go ahead and hardcode the old masquerade subnet.
I will push local changes soon.

Reply:
An alternative can be using different addr label for the new masquerade like `breth0-masq-1` so you can search for the old label `breth0-masq` instead of calculating stuff.

Also if users change configuration you are not going to be able to calculate the old one.

Reply:
Resolving this conversation as I have moved away from adding label to address on external bridge due to iproute not supporting label on IPv6 addresses.